test/e2e/scheduler_predicates.go

```diff
@@ -370,4 +370,65 @@ var _ = Describe("SchedulerPredicates", func() {
 		verifyResult(c, podName, ns, currentlyDeadPods)
 		cleanupPods(c, ns)
 	})
+
+	It("validates that NodeSelector is respected if matching.", func() {
+		// launch a pod to find a node which can launch a pod. We intentionally do
+		// not just take the node list and choose the first of them. Depending on the
+		// cluster and the scheduler it might be that a "normal" pod cannot be
+		// scheduled onto it.
+		By("Trying to launch a pod without a label to get a node which can launch it.")
+		podName := "without-label"
+		_, err := c.Pods(ns).Create(&api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name: podName,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  podName,
+						Image: "gcr.io/google_containers/pause:go",
+					},
+				},
+			},
+		})
+		expectNoError(err)
+		expectNoError(waitForPodRunningInNamespace(c, podName, ns))
+		pod, err := c.Pods(ns).Get(podName)
+		expectNoError(err)
+
+		nodeName := pod.Spec.NodeName
+		err = c.Pods(ns).Delete(podName, api.NewDeleteOptions(0))
+		expectNoError(err)
+
+		By("Trying to relaunch the pod, now with labels.")
+		// TODO(sttts) enable a custom label here, in addition to the hostname
```

Comment:
You can set arbitrary labels manually on the node object on the apiserver. Right now these are not used for scheduling by us:
```
kubectl patch node 192.168.33.60 -p '{"metadata":{"labels":{"foo2":"bar"}}}'  
```